pkg/registry/endpoint/etcd/etcd.go

```diff
@@ -33,7 +33,7 @@ type REST struct {
 }
 
 // NewStorage returns a RESTStorage object that will work against endpoints.
-func NewStorage(h tools.EtcdHelper) *REST {
+func NewStorage(s tools.StorageInterface) *REST {
```

Comment:
Sure - I would like to do it, but I would like to implement the watch in apiserver first. But I'm happy to make this change after that (or during hanging on code review).